### ISSUE TITLE
CLI node segfaults on dev::shh::WhisperPeer::sendMessages()

### ISSUE THREAD

**Issue by jorisbontje** [ethereum/cpp-ethereum#1167]
`EXC_BAD_ACCESS(code=1, address=0x20) in  ; dev::shh::WhisperPeer::sendMessages() + 150`

```
Thread 5:
#0	0x00007fff95d925c1 in _pthread_mutex_lock ()
#1	0x00007fff89961b6f in std::__1::mutex::lock() ()
#2	0x00000001027b3cd1 in dev::shh::WhisperPeer::sendMessages() ()
#3	0x00000001027980af in dev::shh::WhisperHost::doWork() ()
#4	0x00000001027986ec in non-virtual thunk to dev::shh::WhisperHost::doWork() ()
#5	0x0000000103dd4037 in dev::Worker::startWorking()::$_0::operator()() const ()
#6	0x0000000103dd3c29 in void* std::__1::__thread_proxy<std::__1::tuple<dev::Worker::startWorking()::$_0> >(void*) ()
#7	0x00007fff95d94268 in _pthread_body ()
#8	0x00007fff95d941e5 in _pthread_start ()
#9	0x00007fff95d9241d in thread_start ()
```

Nothing special in the logs:
```
*** [ 19:45:10 | miner-1 ] Completed 9e5057b5… 8251d11e… 196635 1
--- [ 19:45:10 | miner-1 ] Completing mine!
*** [ 19:45:10 | miner-1 ] Mined 40a98328… (parent: f2cc6ad6… )
=B= [ 19:45:11 | eth ] Attempting import of 40a98328… ...
=B= [ 19:45:11 | eth ]    Imported and best 118956745 . Has 0 siblings. Route: [ 40a98328…, ]
*** [ 19:45:11 | eth ] New block on chain: Restarting mining operation.
*** [ 19:45:11 | eth ] noteChanged( { 0000000000000000000000000000000000000000000000000000000000000000, 0000000000000000000000000000000000000000000000000000000000000001 } )
-N- [ 19:45:11 | ethsync ] Sending a new block (current is 40a98328dd63e9a301115998fb789bca4be01a7ab212171a0712a2100e50e9c8 , was f2cc6ad6142ce40c6e640f753ea470b8cebe442e0491e1e870fe5d973052aae5 )
--- [ 19:45:11 | <unknown> ] checkWatch 8
--- [ 19:45:11 | <unknown> ] lastPoll updated to 1425149111
*** [ 19:45:11 | <unknown> ] FIRING WATCH 8 1
--- [ 19:45:11 | <unknown> ] checkWatch 9
--- [ 19:45:11 | <unknown> ] lastPoll updated to 1425149111
--- [ 19:45:12 | <unknown> ] checkWatch 8
--- [ 19:45:12 | <unknown> ] lastPoll updated to 1425149112
--- [ 19:45:12 | <unknown> ] checkWatch 9
--- [ 19:45:12 | <unknown> ] lastPoll updated to 1425149112
--- [ 19:45:12 | <unknown> ] startWorking for thread eth
*** [ 19:45:12 | <unknown> ] New transaction {77045e71…/552$5000000000000000000+100000@10000000000000<-cd2a3d9f… #36}
*** [ 19:45:12 | eth ] Additional transaction ready: Restarting mining operation.
*** [ 19:45:12 | eth ] noteChanged( { 0000000000000000000000000000000000000000000000000000000000000000 } )
-N- [ 19:45:13 | p2p | 27 ] Hello: Ethereum(G)/v0.8.5/linux/go1.4.1 V[ 3 ] 4194dbd6… (eth,54)(shh,2) 0
*N* [ 19:45:13 | p2p ] p2p.host.peer.register 4194dbd6…
-N- [ 19:45:13 | p2p | 27 ] Transition! State from Nothing , holding
``` 

This is on OSX with commit e107888. Process started with `eth -j -s $COW_SECRET -m on -f`

**Comment by debris** [comment]
similar here:

```
(lldb) bt
* thread #4: tid = 0xed6e, 0x00007fff8eb519a4 libsystem_pthread.dylib`_pthread_mutex_check_init + 14, stop reason = EXC_BAD_ACCESS (code=1, address=0x20)
    frame #0: 0x00007fff8eb519a4 libsystem_pthread.dylib`_pthread_mutex_check_init + 14
    frame #1: 0x00007fff8eb51621 libsystem_pthread.dylib`_pthread_mutex_lock + 28
    frame #2: 0x00007fff96ec17a9 libc++.1.dylib`std::__1::mutex::lock() + 9
  * frame #3: 0x0000000100aab531 libwhisper.dylib`dev::shh::WhisperPeer::sendMessages() [inlined] std::__1::lock_guard<std::__1::mutex>::lock_guard(this=0x0000000105980648, __m=0x0000000000000020) + 11 at __mutex_base:81
    frame #4: 0x0000000100aab526 libwhisper.dylib`dev::shh::WhisperPeer::sendMessages() [inlined] std::__1::lock_guard<std::__1::mutex>::lock_guard(this=0x0000000105980648, __m=0x0000000000000020) at __mutex_base:81
    frame #5: 0x0000000100aab526 libwhisper.dylib`dev::shh::WhisperPeer::sendMessages(this=0x0000000000000000) + 134 at WhisperPeer.cpp:91
    frame #6: 0x0000000100a9072f libwhisper.dylib`dev::shh::WhisperHost::doWork(this=0x00000001022245d0) + 495 at WhisperHost.cpp:166
    frame #7: 0x0000000100a90d6c libwhisper.dylib`non-virtual thunk to dev::shh::WhisperHost::doWork(this=0x00000001022245e8) + 28 at WhisperHost.cpp:168
    frame #8: 0x0000000101fff867 libdevcore.dylib`dev::Worker::startWorking(this=0x00000001058004a0)::$_0::operator()() const + 455 at Worker.cpp:46
    frame #9: 0x0000000101fff459 libdevcore.dylib`void* std::__1::__thread_proxy<std::__1::tuple<dev::Worker::startWorking(__vp=0x00000001058004a0)::$_0> >(void*) + 393 at memory:2602
    frame #10: 0x00007fff8eb4e899 libsystem_pthread.dylib`_pthread_body + 138
    frame #11: 0x00007fff8eb4e72a libsystem_pthread.dylib`_pthread_start + 137
    frame #12: 0x00007fff8eb52fc9 libsystem_pthread.dylib`thread_start + 13
```

**Comment by subtly** [comment]
WIP: #1366

**Comment by subtly** [comment]
Addressed in #1366.
